src/renderers/dom/shared/ReactDOMTextComponent.js

```diff
@@ -48,6 +48,7 @@ var ReactDOMTextComponent = function(text) {
   this._mountIndex = 0;
   this._closingComment = null;
   this._commentNodes = null;
+  this._isWhitespace = typeof text === 'string' && text.trim().length === 0;
```

Comment:
That's right. `validateDOMNesting` is a noop in production mode and, at the moment, it's the only place that cares about `_isWhitespace`.

So this line should be changed to:
```js
if (__DEV__) {
  this._isWhitespace = typeof text === 'string' && text.trim().length === 0;
}
```

Or did you have something else in mind?